### PR TITLE
overlord/devicestate: try to pick alternative recovery labels during remodel

### PR DESCRIPTION
**Pull request by bboozzoo**
It is possible that a remodel can occur on the same day as the image is built,
in which case try to not fail and pick an alternative recovery system label.

### CODE REVIEW

File: overlord/devicestate/devicestate.go [hunk 1]
```diff
@@ -508,10 +509,28 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
...
-		label := timeNow().Format("20060102")
-		createRecoveryTasks, err := createRecoverySystemTasks(st, label, snapSetupTasks)
-		if err != nil {
-			return nil, err
+		labelBase := timeNow().Format("20060102")
+		labelSuffix := ""
+		var createRecoveryTasks *state.TaskSet
+		// the label is based on date, but if multiple remodels are done
+		// on the same day, we may end up with conflicting labels
+		const maxLabelTries = 5
```
Comment (pedronis) [inline overlord/devicestate/devicestate.go:517]: given that minimal model changes might reuse the same snaps, it unclear how many recovery systems one can stick in the seed even it's not very large.  My preference is to try "<base-label>-1", if that doesn't exist use it, if it exists glob "<base-label>-*", find the max for those where the suffix can be turned into a number, and add one

**Review by pedronis — COMMENTED** [review]
I'm unsure how to pick a maxLabelTries so made a different suggestion

**Review by mardy — COMMENTED** [review]
LGTM, but I'll review it again after Samuele's comment is addressed.

**Review by pedronis — APPROVED** [review]
thanks

**Comment by bboozzoo** [comment]
So I'm playing with the remodel spread test and somehow the system is not booting into the recover mode for the recovery named 20210706-1. Looking at our grub recovery boot script, I believe this snippet https://github.com/snapcore/snapd/blob/master/bootloader/assets/data/grub-recovery.cfg#L35-L38 rejects all systems named like this.

File: overlord/devicestate/devicestate_remodel_test.go [hunk 2]
```diff
@@ -1834,3 +1835,143 @@ func (s *deviceMgrRemodelSuite) TestRemodelUC20RequiredSnapsAndRecoverySystem(c
...
+	// create a conflict with base label
+	err := os.MkdirAll(filepath.Join(boot.InitramfsUbuntuSeedDir, "systems", labelBase), 0755)
+	c.Assert(err, IsNil)
+	for i := 0; i < 5; i++ {
+		// create conflicting labels with numerical suffices
+		l := fmt.Sprintf("%s-%d", labelBase, i)
+		err := os.MkdirAll(filepath.Join(boot.InitramfsUbuntuSeedDir, "systems", l), 0755)
+		c.Assert(err, IsNil)
+	}
+	// and some confusing labels
```
Comment (mvo5) [inline overlord/devicestate/devicestate_remodel_test.go:1920]: Thanks for these

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by bboozzoo** [comment]
recovery grub changes are under control, so I'm droppinng the 'blocked' label

**Review by mardy — APPROVED** [review]
LGTM!

**Comment by bboozzoo** [comment]
The failures are unrelated, @mvo5 @pedronis can you land this PR?

**Pull request merged**